### PR TITLE
ci: fix prerelease comment job to also run when only pages-shared has changed

### PR DESCRIPTION
**Pull request by petebacondarwin**
We were breaking out of the loop too early so only the first artifact was being added to the environment variables.

### CODE REVIEW

File: .github/workflows/write-prerelease-comment.yml [hunk 1]
```diff
@@ -38,9 +38,13 @@ jobs:
             }
 
       - name: "Comment on PR with Wrangler link"
+        env:
+          WORKFLOW_RUN_PR_FOR_WRANGLER: ${{env.WORKFLOW_RUN_PR_FOR_WRANGLER || env.WORKFLOW_RUN_PR_FOR_CLOUDFLARE-PAGES-SHARED}}
+          WORKFLOW_RUN_ID_FOR_WRANGLER: ${{env.WORKFLOW_RUN_ID_FOR_WRANGLER || env.WORKFLOW_RUN_ID_FOR_CLOUDFLARE-PAGES-SHARED}}
```
Comment (RamIdeas) [inline .github/workflows/write-prerelease-comment.yml:43]: Is this intentional or should it be:
```suggestion
          WORKFLOW_RUN_PR_FOR_WRANGLER: ${{env.WORKFLOW_RUN_PR_FOR_WRANGLER || env.WORKFLOW_RUN_PR_FOR_CLOUDFLARE_PAGES_SHARED}}
          WORKFLOW_RUN_ID_FOR_WRANGLER: ${{env.WORKFLOW_RUN_ID_FOR_WRANGLER || env.WORKFLOW_RUN_ID_FOR_CLOUDFLARE_PAGES_SHARED}}
```

File: .github/extract-pr-and-workflow-id.js [hunk 2]
```diff
@@ -17,6 +17,5 @@ for (const artifact of allArtifacts.data.artifacts) {
 			`\nWORKFLOW_RUN_PR_FOR_${packageName}=${match[2]}` +
 				`\nWORKFLOW_RUN_ID_FOR_${packageName}=${context.payload.workflow_run.id}`
```
Comment (RamIdeas) [inline .github/extract-pr-and-workflow-id.js:18]: I see the dashes were intentional

**Review by RamIdeas — APPROVED** [review]

**Comment by rozenmd** [comment]
If this worked, shouldn't we see a PR comment here?

**Comment by petebacondarwin** [comment]
> If this worked, shouldn't we see a PR comment here?

Sadly not. This job only runs on `main`, which is why the bad code got through in the first place.

**Review by rozenmd — APPROVED** [review]
In that case, LGTM

**Pull request merged**